### PR TITLE
Issue 544 -- Add refDepthMin and refDepthMax options

### LINKED ISSUE #544: Allow a user to set a depth for circular $refs, or a $ref depth minimum and maximum
As I understand, [currently](https://github.com/json-schema-faker/json-schema-faker/blob/develop/src/lib/core/run.js#L139), for circular `$ref`s, JSF will randomly select a depth to render, between 1 and 3. 

I have code working, based on https://github.com/json-schema-faker/json-schema-faker/pull/543 (it needs the `minItems` and `maxItems` arguments in order to work), to add a "`refDepth`" API option.

Before I open a PR for this, I'd just like to check, is this an idea you'd be open to? If so, would you prefer a single `refDepth` option, or a `refDepthMin` and `refDepthMax`?

### PR DESCRIPTION
**Pull request by publicus**
This PR resolves #544. It adds two new options (with tests), `refDepthMin` and `refDepthMax`, allowing control over how deep circular references are rendered. For example, by setting `refDepthMin` and `refDepthMax` both to 4, one can get a circular array that will always render four levels deep, like this:

```
{
  "collection": {
    "a": {
      "subA": [
        {
          "subA": [
            {
              "subA": [
                {
                  "subA": []
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

As was the case before this change, `refDepthMin` defaults to 0, and `refDepthMax` defaults to 3.

### CODE REVIEW

File: src/lib/core/run.js [hunk 1]
```diff
@@ -136,7 +138,7 @@ function run(refs, schema, container) {
 
       if (typeof sub.$ref === 'string') {
         // increasing depth only for repeated refs seems to be fixing #258
-        if (sub.$ref === '#' || (lastRef === sub.$ref && ++depth > random.number(0, 3))) {
+        if (sub.$ref === '#' || (lastRef === sub.$ref && ++depth > random.number(refDepthMin, refDepthMax) - 1)) {
```
Comment (publicus) [inline src/lib/core/run.js]: I can remove the `-1` here if desired -- it depends on how one defines "depth" here -- whether the maximum depth should be empty, or whether it should be fully rendered, with one additional, one-level-deeper, empty element in it.

With `-1`, with `refDepthMin` and `refDepthMax` both set to 4:

```
{
  "collection": {
    "a": {
      "subA": [
        {
          "subA": [
            {
              "subA": [
                {
                  "subA": []
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

Without `-1`, with `refDepthMin` and `refDepthMax` both set to 4:

```
{
  "collection": {
    "a": {
      "subA": [
        {
          "subA": [
            {
              "subA": [
                {
                  "subA": [
                    {
                      "subA": []
                    }
                  ]
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

Comment (publicus) [reply src/lib/core/run.js]: The first of those looks preferable to me, of those two, so I included `-1` in the PR.

Comment (pateketrueke) [reply src/lib/core/run.js]: Yeah, I liked, but can you do me a favor?

Please validate both values, ensure `refDepthMin` is always less-or-equal (?) than `refDepthMax` and then, just subtract the -1 as desired, e.g.

```js
const maxDepth = random.number(refDepthMin, refDepthMax) - 1;
// ...
if ( ... ++depth > maxDepth)
// ...
```

Comment (publicus) [reply src/lib/core/run.js]: Sure! I've pushed several commits, ending with d9537331a, just now. Do those changes address what you're looking for?

**Pull request merged**